src/adapter/src/coord.rs

```diff
@@ -788,10 +783,39 @@ impl<S: Append + 'static> Coordinator<S> {
         }
     }
 
-    #[allow(dead_code)]
-    async fn create_storage_export(&mut self, id: GlobalId, sink: &Sink) {
+    async fn create_storage_export(
+        &mut self,
+        id: GlobalId,
+        sink: &Sink,
+        connection: StorageSinkConnection,
+    ) -> Result<(), AdapterError> {
+        // Validate `sink.from` is in fact a storage collection
+        self.controller.storage().collection(sink.from)?;
+
+        // The AsOf is used to determine at what time to snapshot reading from the persist collection.  This is
+        // primarily relevant when we do _not_ want to include the snapshot in the sink.  Choosing now will mean
+        // that only things going forward are exported.
+        let timeline = self
```

Comment:
nit: feels like this whole block could be factored out into a method. But we can also leave this for now because we want to merge.